src/AppInstallerRepositoryCore/Public/winget/RepositorySearch.h

```diff
@@ -243,8 +233,16 @@ namespace AppInstaller::Repository
 
         // The channel.
         Utility::NormalizedString Channel;
+
+        bool operator<(const PackageVersionKey& other) const
+        {
+            // Sort using only the version and channel.
+            // The order for the sources depends on the context.
+            return Utility::VersionAndChannel({ Version }, { Channel }) < Utility::VersionAndChannel({ other.Version }, { other.Channel });

```

Comment:
This code is assuming versions across sources use same semantic. I remember seeing packages that use different versions for Store source and other sources. Should we not allow version comparison if it's from different sources?